### PR TITLE
Refac(T32650): Vue migration: convert DpTableRow into a template-based component

### PR DESCRIPTION
**Pull request by sakutademos**
**Ticket:** https://yaits.demos-deutschland.de/T32650

**Description:** This PR converts a functional Vue` DpTableRow` component into a template-based component. 

**Important:** This PR depends on https://github.com/demos-europe/demosplan-ui/pull/230.

### CODE REVIEW

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: src/components/core/DpDataTable/DpTableRow.vue [hunk 1]
```diff
@@ -1,13 +1,105 @@
...
+    </td>
+
+    <td
+      v-for="(field, idx) in fields"
+      :key="`${field}:${idx}`"
+      :class="{ 'c-data-table__resizable': isTruncatable }"
+      :data-col-idx="`${idx}`">
+      <div
+        v-if="isTruncatable"
+        :class="wrapped ? 'c-data-table__resizable--wrapped overflow-word-break' : 'c-data-table__resizable--truncated overflow-word-break'"
```
Comment (salisdemos) [inline src/components/core/DpDataTable/DpTableRow.vue:39]: nitpick
```suggestion
        class="overflow-word-break"
        :class="wrapped ? 'c-data-table__resizable--wrapped' : 'c-data-table__resizable--truncated'"
```

File: src/components/core/DpDataTable/DpTableRow.vue [hunk 2]
```diff
@@ -1,13 +1,105 @@
...
+        </slot>
+      </div>
+      <template v-else>
+        <slot
+          :name="field"
+          v-bind="item">
+          <span
+            v-if="searchTerm && item[field]"
+            v-html="highlighted(field)" />
+          <span v-text="item[field]" v-else />
```
Comment (salisdemos) [inline src/components/core/DpDataTable/DpTableRow.vue:59]:
```suggestion
          <span
            v-text="item[field]"
            v-else />
```

**Review by salisdemos — CHANGES_REQUESTED** [review]
Just nitpicking

File: src/components/core/DpDataTable/DpTableRow.vue (hunk 2 above)
Comment (sakutademos) [reply src/components/core/DpDataTable/DpTableRow.vue]: done here https://github.com/demos-europe/demosplan-ui/pull/233/commits/2b39dcb48f5fc71f8fa4e4e36395be699b6955bb

File: src/components/core/DpDataTable/DpTableRow.vue (hunk 1 above)
Comment (sakutademos) [reply src/components/core/DpDataTable/DpTableRow.vue]: thx! done here https://github.com/demos-europe/demosplan-ui/pull/233/commits/2b39dcb48f5fc71f8fa4e4e36395be699b6955bb

**Review by salisdemos — APPROVED** [review]

**Pull request merged**